### PR TITLE
1. Add API creds into KeyVault so that RP can read it

### PR DESCRIPTION
**Pull request by ross-p-smith**
# PR for issue #220

## What is being addressed

This PR places the credentials for the API into the core KeyVault so that the Resource Provider can read them so that it can communicate with the Auth tenant. This is part 1 of multiple PR's

## How is this addressed

- Add 3 new secrets into KeyVault
- Retrofitted the api-app app config so that it uses these key vault secrets rather than clear text values.
- Add the keyvault name as an environment variable to the VMSS instance.
- Utilise a porter plugin that automatically reads from KeyVault (from the name above) 
- Use a new credential set in Porter to retrieve the necessary secrets from the Vault
- #1701 will continue this work

### CODE REVIEW

**Comment by tamirkamara** [comment]
Why does the RP need the API app credentials?

**Comment by ross-p-smith** [comment]
> Why does the RP need the API app credentials?

So that when `auto_app_registration` is set to true, it is able to create the App Registration for the workspace.

File: resource_processor/resources/commands.py [hunk 1]
```diff
@@ -59,7 +59,9 @@ async def build_porter_command(config, logger, msg_body, custom_action=False):
                     f"{'invoke --action ' if custom_action else ''}"
                     f"{msg_body['action']} \"{installation_id}\" "
                     f" --reference {config['registry_server']}/{msg_body['name']}:v{msg_body['version']}"
-                    f" {porter_parameters} --cred ./vmss_porter/azure.json --allow-docker-host-access --force"
+                    f" {porter_parameters} --allow-docker-host-access --force"
+                    f" --cred ./vmss_porter/azure.json"
+                    f" --cred ./vmss_porter/api-creds.json"
```
Comment (tamirkamara) [inline resource_processor/resources/commands.py:64]: workspace_creation... instead of api-creds?

File: resource_processor/run.sh [hunk 2]
```diff
@@ -9,8 +9,14 @@ if [[ -z "${MGMT_STORAGE_ACCOUNT_NAME}" ]]; then
...
 fi
 
+if [[ -z "${KEY_VAULT_NAME}" ]]; then
+  >&2 echo "Environment variable for Key Vault name missing"
+fi
+
+# Documentation here: - https://github.com/vdice/porter-bundles/tree/master/azure-keyvault
 cat > /root/.porter/config.toml << EOF
 default-storage = "azurestorage"
+default-secrets = "api"
```
Comment (tamirkamara) [inline resource_processor/run.sh:19]: still need to be api?

File: resource_processor/run.sh [hunk 3]
```diff
@@ -20,6 +26,13 @@ plugin = "azure.table"
 [storage.config]
 account="${MGMT_STORAGE_ACCOUNT_NAME}"
 resource-group="${MGMT_RESOURCE_GROUP_NAME}"
+
+[[secrets]]
+name = "api"
```
Comment (tamirkamara) [inline resource_processor/run.sh:31]: still need to be api?

File: resource_processor/vmss_porter/api-creds.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (tamirkamara) [inline resource_processor/vmss_porter/api-creds.json:1]: Should the file be renamed?

**Review by tamirkamara — APPROVED** [review]

**Comment by ross-p-smith** [comment]
/test
(manually deleted environment)

**Comment by ross-p-smith** [comment]
Merged this branch into the later branch and the tests pass on that branch.
/test-force-approve

**Comment by ross-p-smith** [comment]
/test-force-approve

(Merged this branch into another and ran tests there)

**Pull request merged**